consensus/istanbul/backend/engine_test.go

```diff
@@ -654,70 +658,117 @@ func TestSnapshot(t *testing.T) {
 		{
 			[]uint64{5000000, 5000000, 5000000, 5000000},
 			false,
+			false,
 			[]int{0, 1, 2, 3},
 			[]int{},
 		},
 		{
 			[]uint64{5000000, 5000000, 5000000, 6000000},
 			false,
+			false,
 			[]int{0, 1, 2, 3},
 			[]int{},
 		},
 		{
 			[]uint64{5000000, 5000000, 6000000, 6000000},
 			false,
+			false,
 			[]int{0, 1, 2, 3},
 			[]int{},
 		},
 		{
 			[]uint64{5000000, 6000000, 6000000, 6000000},
 			false,
+			false,
 			[]int{0, 1, 2, 3},
 			[]int{},
 		},
 		{
 			[]uint64{6000000, 6000000, 6000000, 6000000},
 			false,
+			false,
 			[]int{0, 1, 2, 3},
 			[]int{},
 		},
 		// The following testcases are the ones after istanbul incompatible change
 		{
 			[]uint64{5000000, 5000000, 5000000, 5000000},
 			true,
+			false,
 			[]int{0, 1, 2, 3},
 			[]int{},
 		},
 		{
 			[]uint64{5000000, 5000000, 5000000, 6000000},
 			true,
+			false,
 			[]int{3},
 			[]int{0, 1, 2},
 		},
 		{
 			[]uint64{5000000, 5000000, 6000000, 6000000},
 			true,
+			false,
 			[]int{2, 3},
 			[]int{0, 1},
 		},
 		{
 			[]uint64{5000000, 6000000, 6000000, 6000000},
 			true,
+			false,
 			[]int{1, 2, 3},
 			[]int{0},
 		},
 		{
 			[]uint64{6000000, 6000000, 6000000, 6000000},
 			true,
+			false,
 			[]int{0, 1, 2, 3},
 			[]int{},
 		},
 		{
 			[]uint64{5500001, 5500000, 5499999, 0},
 			true,
+			false,
 			[]int{0, 1},
 			[]int{2, 3},
 		},
+		// The following testcases are the ones for testing governing node in single mode
```

Comment:
Is the first element of `stakingAmounts` slice the governing node's staking amount?

Reply:
@yoomee1313 Yes, I have added the comments for governing node.